lib/jsdom/browser/Window.js

```diff
@@ -76,6 +77,16 @@ function Window(options) {
 
   ///// PRIVATE DATA PROPERTIES
 
+  if (options.resourceLoader) {
+    if (!(options.resourceLoader instanceof ResourceLoader)) {
+      throw new RangeError("options.ResourceLoader must implement ResourceLoader");
+    }
+
+    this._resourceLoader = options.resourceLoader;
+  } else {
```

Comment:
This should never happen, right? Since you already did the defaulting in api.js. Note that the VirtualConsole pattern has a comment saying it is not necessary in the new API :).

Reply:
You mean the `else` part right? At least we still need the type checking.